### PR TITLE
[ros2] grid_map_octomap port

### PR DESCRIPTION
**Pull request by Marwan99**
* Linted files
* Updated package.xml & CMakeLists.txt
* Added ocotmap as a dependency to ros2_dependencies.repos

### CODE REVIEW

**Comment by SteveMacenski** [comment]
https://github.com/OctoMap/octomap/issues/300

lets wait a minute to see if they release so we don't need to have it in our CI config. 

Otherwise LGTM once CI passes

**Comment by Marwan99** [comment]
The empty stderr from octomap is not really very helpful :'). I get this empty error building octomap for the first time and the error is gone by simply rebuilding again.

image

@SteveMacenski help please!

**Comment by Marwan99** [comment]
Actually, let me try one more thing.

File: grid_map_octomap/CMakeLists.txt [hunk 1]
```diff
@@ -1,69 +1,36 @@
...
-  ${catkin_LIBRARIES}
-  ${OCTOMAP_LIBRARIES}
+ament_target_dependencies(${PROJECT_NAME}
+  ${dependencies}
 )
 
-add_dependencies(${PROJECT_NAME}
-  ${catkin_EXPORTED_TARGETS}
-)
+# add_dependencies(${PROJECT_NAME}
```
Comment (maximilianwulf) [inline grid_map_octomap/CMakeLists.txt:31]: Could this also be deleted?

**Comment by SteveMacenski** [comment]
@Marwan99 let me know what you want me to review

**Comment by Marwan99** [comment]
@SteveMacenski I think it's already ready.

**Comment by SteveMacenski** [comment]
https://github.com/ros/rosdistro/blob/master/foxy/distribution.yaml#L1129

Foxy now has octomap, try removing the ros2_dependency on it

**Comment by Marwan99** [comment]
rosdep is throwing the following error:
``` 
ERROR: the following rosdeps failed to install
  apt: command [apt-get install -y ros-foxy-octomap] failed
 ```
@SteveMacenski Any suggestions?

**Comment by SteveMacenski** [comment]
I think it just needs a foxy sync https://discourse.ros.org/t/preparing-for-foxy-sync-and-patch-release-2020-07-10/15265 which will happen in 4 days. I think we can wait until then and just move onto the next pkg

**Comment by Marwan99** [comment]
I think that should be ready now. @SteveMacenski Can you please re-run CI? (I am assuming you have write permission on circleci)

**Comment by SteveMacenski** [comment]
retriggered

**Comment by Marwan99** [comment]
CI passed!

**Pull request merged**